src/pages/TokenDetails/index.tsx

```diff
@@ -80,8 +81,29 @@ export default function TokenDetails() {
   const token = useToken(tokenAddress)
 
   const tokenWarning = tokenAddress ? checkWarning(tokenAddress) : null
-  /* network balance handling */
+  const [continueSwap, setContinueSwap] = useState<{ resolve: (value: boolean | PromiseLike<boolean>) => void }>()
+
+  const navigate = useNavigate()
+
+  const onSwapReview = useCallback(() => {
+    return new Promise<boolean>(function (resolve, reject) {
+      setContinueSwap({ resolve })
+    })
+  }, [])
+
+  const resolveSwap = useCallback(
```

Comment:
nit: as a callback, rename this to `onResolveSwap`. Alternatively, define separately callbacks for `onCancel` and `onContinue`.